modules/ui/org.eclipse.fx.ui.preferences/src/main/java/org/eclipse/fx/ui/preferences/PreferenceUI.java

```diff
@@ -93,11 +93,13 @@
 
 	private final PreferencePageFactory factory;
 
-	protected String currentFilter = "";
+	protected static String currentFilter = "";
```

Comment:
why static? This is not a good idea

Reply:
It has to be static in order to access it from the constructor. I reverted the change and added back the filter and comparator methods that the subclasses can override.